cores/rp2040/SerialPIO.cpp

```diff
@@ -46,12 +46,17 @@ static pio_program_t *pio_make_uart_prog(int repl, const pio_program_t *pg) {
     return p;
 }
 
-static PIOProgram *_getTxProgram(int bits) {
-    auto f = _txMap.find(bits);
-    if (f == _txMap.end()) {
-        pio_program_t * p = pio_make_uart_prog(bits, &pio_tx_program);
-        _txMap.insert({bits, new PIOProgram(p)});
-        f = _txMap.find(bits);
+static PIOProgram *_getTxProgram(int bits, bool inverted) {
+		int key = bits * (inverted ? -1 : 1);
+    auto f = _txMap.find(key);
+    if (f == _txMap.end()) { 
+				pio_program_t * p;
+				if (inverted)
```

Comment:
Not a fan of ternarys?  The `astyle` will blow this up to 6 lines or so because it requires `{}`s around conditions.  I've been burned in past lives before w/o them, so it's part of the core style rules now.
One-liner:  `pio_make_uart_prog(bits, _inverted ? &pio_tx_inv_program : pio_tx_program);`